platform/ios/src/MGLMapboxEvents.m

```diff
@@ -429,10 +424,15 @@ - (void)startUpdatingLocation {
     }
 }
 
+static BOOL turnstileEventSent = NO;
+
 - (void) pushTurnstileEvent {
 
-    __weak MGLMapboxEvents *weakSelf = self;
+    // Turnstile events only need to be sent up once per application launch
+    if (turnstileEventSent) { return; }
```

Comment:
We should make a call on `if () return` style in MGLMapboxEvents and harmonize the entire thing.